src/object/cli_obj.c

```diff
@@ -3085,6 +3086,11 @@ obj_shard_comp_cb(struct shard_auxi_args *shard_auxi,
 	tse_task_t		*task = obj_auxi->obj_task;
 	int			ret = task->dt_result;
 
+	if (shard_auxi == NULL) {
+		iter_arg->retry = false;
+		return ret;
+	}
```

Comment:
is this to fix the hang in samir's test?
seems not an ideal fix although, as this will cause the key query can not be retried for some cases?
seems can add "struct shard_query_key_args" to struct obj_auxi_args' union, and change dc_obj_query_key to use that shard_auxi?

Reply:
this is to fix the panic in some cases if shard_auxi == NULL, the following code will cause panic.

I am not sure what is the hang yet.  oh, this patch is not for resolving the key query, I thought you are working on it?

Reply:
ok, I'll refine the key query things later. thx